### ISSUE TITLE
Chat-hide no longer works in BBB 2.2.31

### ISSUE THREAD

**Issue by ichdasich** [aau-zid/BigBlueButton-liveStreaming#109]
**Describe the bug**
Running against BBB 2.2.31, chat hide does not work. When the stream starts, the chat is hidden at first, but reappears quickly thereafter. When running against BBB 2.2.27, this issue does not occur.

**To Reproduce**
1. Start container against BBB 2.2.31
2. Watch web-stream
3. See chat briefly disappearing, and then popping in again

**Expected behavior**
Chat remains hidden when configured to be hidden.

**Logs**
```
one
Attaching to liveStreaming
liveStreaming   | Skipping nsswrapper setup - already initialized
liveStreaming   | Starting pulseaudio...
liveStreaming   | Waiting for pulseaudio to start...
liveStreaming   | Skipping nsswrapper setup - already initialized
liveStreaming   | INFO:root:Starting browser!!
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session {"capabilities": {"firstMatch": [{}], "alwaysMatch": {"browserName": "chrome", "platformName": "any", "goog:chromeOptions": {"excludeSwitches": ["enable-automation"], "prefs": {"intl.accept_languages": "en_US.UTF-8"}, "extensions": [], "args": ["--disable-infobars", "--no-sandbox", "--kiosk", "--window-size=1280,720", "--window-position=0,0", "--start-fullscreen"]}}}, "desiredCapabilities": {"browserName": "chrome", "version": "", "platform": "ANY", "goog:chromeOptions": {"excludeSwitches": ["enable-automation"], "prefs": {"intl.accept_languages": "en_US.UTF-8"}, "extensions": [], "args": ["--disable-infobars", "--no-sandbox", "--kiosk", "--window-size=1280,720", "--window-position=0,0", "--start-fullscreen"]}}}
liveStreaming   | DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): 127.0.0.1:35523
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session HTTP/1.1" 200 717
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | INFO:root:Open BBB and hide elements!!
liveStreaming   | INFO:root:get_join_url...  
liveStreaming   | INFO:root:https://REDACTED/bigbluebutton/api/join?meetingID=REDACTED&fullName=Live&password=REDACTED&userdata-bbb_auto_join_audio=true&userdata-bbb_enable_video=true&userdata-bbb_listen_only_mode=true&userdata-bbb_force_listen_only=true&userdata-bbb_skip_check_audio=true&joinViaHtml5=true&checksum=REDACTED
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/url {"url": "https://REDACTED/bigbluebutton/api/join?meetingID=REDACTED&fullName=Live&password=REDACTED&userdata-bbb_auto_join_audio=true&userdata-bbb_enable_video=true&userdata-bbb_listen_only_mode=true&userdata-bbb_force_listen_only=true&userdata-bbb_skip_check_audio=true&joinViaHtml5=true&checksum=REDACTED"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/url HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 259
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 200 88
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/elements {"using": "xpath", "value": "//span[contains(@class,\"success\")]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/elements HTTP/1.1" 200 90
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/ed5f1550-ba56-4d2e-99ce-6e6a23dbca6a/click {"id": "ed5f1550-ba56-4d2e-99ce-6e6a23dbca6a"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/ed5f1550-ba56-4d2e-99ce-6e6a23dbca6a/click HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "css selector", "value": ".ReactModal__Overlay"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 404 250
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element {"using": "css selector", "value": "[id=\"message-input\"]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element HTTP/1.1" 200 88
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/elements {"using": "css selector", "value": "[aria-label=\"Send message\"]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/elements HTTP/1.1" 200 90
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:GET http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/7eec3234-a235-4ad1-8021-7f2faab2d2a9/enabled {"id": "7eec3234-a235-4ad1-8021-7f2faab2d2a9"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "GET /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/7eec3234-a235-4ad1-8021-7f2faab2d2a9/enabled HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:GET http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/30bbe885-50e9-4e63-87e9-875e465231da/enabled {"id": "30bbe885-50e9-4e63-87e9-875e465231da"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "GET /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/30bbe885-50e9-4e63-87e9-875e465231da/enabled HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/7eec3234-a235-4ad1-8021-7f2faab2d2a9/value {"text": "This meeting is streamed to: REDACTED", "value": ["REDACTED"], "id": "7eec3234-a235-4ad1-8021-7f2faab2d2a9"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/7eec3234-a235-4ad1-8021-7f2faab2d2a9/value HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/30bbe885-50e9-4e63-87e9-875e465231da/click {"id": "30bbe885-50e9-4e63-87e9-875e465231da"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/30bbe885-50e9-4e63-87e9-875e465231da/click HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/elements {"using": "css selector", "value": "[id=\"chat-toggle-button\"]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/elements HTTP/1.1" 200 90
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:GET http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/d881ceb0-fe19-4f7d-984c-112487e358c6/enabled {"id": "d881ceb0-fe19-4f7d-984c-112487e358c6"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "GET /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/d881ceb0-fe19-4f7d-984c-112487e358c6/enabled HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/d881ceb0-fe19-4f7d-984c-112487e358c6/click {"id": "d881ceb0-fe19-4f7d-984c-112487e358c6"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/d881ceb0-fe19-4f7d-984c-112487e358c6/click HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/elements {"using": "css selector", "value": "button[aria-label=\"Users and messages toggle\"]"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/elements HTTP/1.1" 200 90
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:GET http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/55938847-4673-4130-a0c9-cb9eb017d7e3/enabled {"id": "55938847-4673-4130-a0c9-cb9eb017d7e3"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "GET /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/55938847-4673-4130-a0c9-cb9eb017d7e3/enabled HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/element/55938847-4673-4130-a0c9-cb9eb017d7e3/click {"id": "55938847-4673-4130-a0c9-cb9eb017d7e3"}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/element/55938847-4673-4130-a0c9-cb9eb017d7e3/click HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync {"script": "document.querySelector('[aria-label=\"Users and messages toggle\"]').style.display='none';", "args": []}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync {"script": "document.querySelector('[aria-label=\"Options\"]').style.display='none';", "args": []}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync {"script": "document.querySelector('[aria-label=\"Actions bar\"]').style.display='none';", "args": []}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:POST http://127.0.0.1:35523/session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync {"script": "document.getElementById('container').setAttribute('style','margin-bottom:30px');", "args": []}
liveStreaming   | DEBUG:urllib3.connectionpool:http://127.0.0.1:35523 "POST /session/8e3ca1b7d4cef7cf88054651a2fcc446/execute/sync HTTP/1.1" 200 14
liveStreaming   | DEBUG:selenium.webdriver.remote.remote_connection:Finished Request
liveStreaming   | DEBUG:root:Preparing to execute 'ffmpeg -thread_queue_size "1024" -f x11grab -draw_mouse 0 -s 1280x720  -i :122 -thread_queue_size "1024" -f pulse -i default -ac 2         -c:a aac -b:a 160k -ar 44100         -threads "0"         -c:v libx264 -x264-params "nal-hrd=cbr" -profile:v high -level:v 4.2 -vf format=yuv420p         -b:v "4000k" -maxrate "4000k" -minrate "2000k" -bufsize "8000k" -g 60         -preset ultrafast          -f flv -flvflags no_duration_filesize "rtmp://10.23.42.2:1935/live/tob-cre-ye9-1dv"'
liveStreaming   | INFO:root:streaming meeting...
liveStreaming   | ffmpeg version 4.3.2-0york0~18.04 Copyright (c) 2000-2021 the FFmpeg developers
liveStreaming   |   built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)
liveStreaming   |   configuration: --prefix=/usr --extra-version='0york0~18.04' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libzimg --enable-pocketsphinx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
```

**Comment by ichdasich** [comment]
Ok, this bugged me;

After looking at this some more, it seems like BBB gets quirky if one tries to hide the chat while the 'cooldowns' for the 'joined audio' and 'default presentation' pop-ups are still running. I could confirm the same behavior (hidden chat pops back up while those are still counting down) with my desktop browser.

Adding a 10 second sleep before hiding chat helps (but is a hot-fix; This should be fixed in BBB). Nevertheless, i made a pull request for this.

**Comment by ichdasich** [comment]
Resolved by #110
